src/proto/FunctionRpc.proto

```diff
@@ -262,6 +267,51 @@ message RpcFunctionMetadata {
   bool is_proxy = 7;
 }
 
+// Host tells worker it is ready to receive metadata
+message WorkerFunctionMetadataRequest {
+  // base directory for function app
+  string directory = 1;
+}
+
+// Worker sends function metadata back to host
+message WorkerFunctionMetadataResponse {
+  // list of function indexing responses
+  repeated WorkerFunctionMetadata results = 1;
+
+  // status of overall metadata request
+  StatusResult overall_status = 2;
+}
+
+// Message type that holds function metadata information
+message WorkerFunctionMetadata {
+  // function name
+  string name = 4;
+
```

Comment:
instead of defining a new type - Reuse `RpcFunctionMetadata` and augment any new fields as needed

Reply:
Adding new fields to RpcFunctionMetadata shouldn't affect anything else that was already using RpcFunctionMetadata right? Just want to make sure.